packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1691,6 +1691,22 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
 }
 
 /// A modal route that overlays a widget over the current route.
+///
+/// When [barrierDismissible] is true, the user can dismiss the route by tapping
```

Comment:
IT looks like this is documented on ModalRoute.barrierDismissible, can you update those docs and create a template to re-use here?

Reply:
Great idea! I just updated ModalRoute docs, I noticed docs for `ModalRoute.barrierDismissible` is a template itself but it has too many details that we don't need here, and turned the part we need into a smaller template and mentioned macro here.

Reply:
Fab!

Reply:
Actually, I had to revert the smaller template as it was nested, after updating the PR, the analyzer throws.
So I simply added the entire `ModalRoute.barrierDismissible` template to PopupRoute docs